services/dynamic-scheduler/tests/unit/test_services_services_tracker__notfier.py

```diff
@@ -0,0 +1,261 @@
+from contextlib import AsyncExitStack, asynccontextmanager
+from unittest.mock import AsyncMock
+
+import pytest
+import socketio
+from aiohttp import web
+from aiohttp.test_utils import TestServer
+from faker import Faker
+from fastapi import FastAPI
+from fastapi.encoders import jsonable_encoder
+from models_library.api_schemas_directorv2.dynamic_services import DynamicServiceGet
+from models_library.api_schemas_dynamic_scheduler.socketio import (
+    SOCKET_IO_SERVICE_STATUS_EVENT,
+)
+from models_library.api_schemas_webserver.projects_nodes import NodeGet, NodeGetIdle
+from models_library.users import GroupID
+from pydantic import NonNegativeInt, parse_obj_as
+from pytest_mock import MockerFixture
+from pytest_simcore.helpers.typing_env import EnvVarsDict
+from servicelib.socketio_utils import cleanup_socketio_async_pubsub_manager
+from servicelib.utils import logged_gather
+from settings_library.rabbit import RabbitSettings
+from settings_library.redis import RedisSettings
+from simcore_service_dynamic_scheduler.core.settings import ApplicationSettings
+from simcore_service_dynamic_scheduler.services.services_tracker._notifier._core import (
+    publish_message,
+)
+from simcore_service_dynamic_scheduler.services.services_tracker.api import (
+    ServicesTracker,
+    get_services_tracker,
+)
+from socketio import AsyncAioPikaManager, AsyncServer
+from tenacity import AsyncRetrying
+from tenacity.stop import stop_after_attempt
+from tenacity.wait import wait_fixed
+from yarl import URL
+
+pytest_simcore_core_services_selection = [
+    "rabbit",
+    "redis",
+]
+pytest_simcore_ops_services_selection = []
+
+
+@pytest.fixture
+def app_environment(
+    app_environment: EnvVarsDict,
+    redis_service: RedisSettings,
+    rabbit_service: RabbitSettings,
+) -> EnvVarsDict:
+    return app_environment
+
+
+@pytest.fixture
+async def services_tracker(app: FastAPI) -> AsyncIterable[ServicesTracker]:
+    tracker = get_services_tracker(app)
+    yield tracker
+    # cleanup distributed cache between tests to avoid flakiness
+    await tracker.service_status_cache.clear()
+
+
+@pytest.fixture
+async def socketio_server(app: FastAPI) -> AsyncIterable[AsyncServer]:
+    # Same configuration as simcore_service_webserver/socketio/server.py
+    settings: ApplicationSettings = app.state.settings
+    server_manager = AsyncAioPikaManager(url=settings.DYNAMIC_SCHEDULER_RABBITMQ.dsn)
+
+    server = AsyncServer(
+        async_mode="aiohttp",
+        engineio_logger=True,
+        client_manager=server_manager,
+    )
+
+    yield server
+
+    await cleanup_socketio_async_pubsub_manager(server_manager)
+
+
+@pytest.fixture
+async def web_server(
+    socketio_server: AsyncServer, aiohttp_unused_port: Callable
+) -> AsyncIterator[URL]:
+    """
+    this emulates the webserver setup: socketio server with
+    an aiopika manager that attaches an aiohttp web app
+    """
+    aiohttp_app = web.Application()
+    socketio_server.attach(aiohttp_app)
```

Comment:
question: why do you care to simulate the webserver here? these are unit tests of the dynamic scheduler right?
in case you want to check the webserver, then do an integration test. otherwise I would not complicate stuff by adding aiohttp stuff here.